ddtrace/llmobs/_llmobs.py

```diff
@@ -124,26 +124,10 @@ def _child_after_fork(self):
         self._evaluator_runner = self._evaluator_runner.recreate()
         self._trace_processor._span_writer = self._llmobs_span_writer
         self._trace_processor._evaluator_runner = self._evaluator_runner
-        tracer_filters = self.tracer._filters
-        if not any(isinstance(tracer_filter, LLMObsTraceProcessor) for tracer_filter in tracer_filters):
-            tracer_filters += [self._trace_processor]
-        self.tracer.configure(settings={"FILTERS": tracer_filters})
-        try:
-            self._llmobs_span_writer.start()
-            self._llmobs_eval_metric_writer.start()
-        except ServiceStatusError:
-            log.debug("Error starting LLMObs writers after fork")
-
-        try:
-            self._evaluator_runner.start()
-        except ServiceStatusError:
-            log.debug("Error starting evaluator runner after fork")
+        if self.enabled:
+            self._start_service()
```

Comment:
If we took the tracer processor check/add step out of `_start_service()`, this means for cases where llmobs is already enabled and then forked into child processes, the tracer will be recreated and have its tracer filters be reset to default (i.e. no llmobs trace processor), meaning the tracer will be disconnected to the recreated llmobs writer, i.e. reverting back to this [issue](https://github.com/DataDog/dd-trace-py/pull/10249/files). Moving the tracer filter setting steps to `enable()` won't be sufficient as this won't be automatically run on fork.

I think the easiest solution is to just move the steps to check/set tracer filters into `_start_service()`, and call `_start_service()` here as you've done already. WDYT?

Side note: I think the existing tests may be insufficient, as @lievan also ran into a similar forking issue where the tests passed but trying it in a celery+gunicorn test app reproduced the errors. Maybe worth verifying?

Reply:
with the move of the filter adding back to _start_service() we should be good again